### PR TITLE
Use official OTLP types in api_v3 and avoid triple-marshaling

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?
- Part of #5079
- Avoid triple marshaling due to our use of internally generated OTLP proto types, which prevents us from directly using the output of jaeger->otlp conversion from collector contrib.

## Description of the changes
- 🛑 breaking: the JSON format is changed to match [OTLP-JSON][otlp-json], specifically
  - the trace & span IDs are returned as hex-encoded strings, not base64 as required by Proto-JSON
  - enums are returned as integers, not strings
- Use Proposal 1 from https://github.com/open-telemetry/opentelemetry-collector/issues/9233#issuecomment-1883868433
- API-V3 proto is already declared to use official OTLP types; remove the overrides to our internally generated OTLP proto types
- Replace `SpansResponseChunk` with official `otlp.TracesData`, but override it internally to use a custom type
- Depends on https://github.com/jaegertracing/jaeger-idl/pull/103

## How was this change tested?
- Unit tests

[otlp-json]: https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#json-protobuf-encoding

### CODE REVIEW

File: Makefile.Protobuf.mk [hunk 1]
```diff
@@ -146,14 +146,13 @@ proto-otel: proto-prepare-otel
...
+# which at runtime is mapped to a custom type in cmd/query/app/internal/api_v3/traces.go
+API_V3_PATH=cmd/query/app/internal/api_v3
 .PHONY: proto-api-v3
 proto-api-v3:
-	$(call print_caption, Enriching OpenTelemetry Protos into $(PATCHED_OTEL_PROTO_DIR))
-	rm -rf $(PATCHED_OTEL_PROTO_DIR)/*
-	cp -R idl/opentelemetry-proto/* $(PATCHED_OTEL_PROTO_DIR)
-	find $(PATCHED_OTEL_PROTO_DIR) -name "*.proto" | xargs -L 1 $(SED) -i 's+go.opentelemetry.io/proto/otlp+github.com/jaegertracing/jaeger/proto-gen/otel+g'
-
-	$(call proto_compile, proto-gen/api_v3, idl/proto/api_v3/query_service.proto, -I$(PATCHED_OTEL_PROTO_DIR))
```
Comment (yurishkuro) [inline Makefile.Protobuf.mk:159]: * enriching is no longer needed
* types are generated into an internal module now

File: cmd/all-in-one/all_in_one_test.go [hunk 2]
```diff
@@ -180,9 +179,11 @@ func getServicesAPIV3(t *testing.T) {
 	require.NoError(t, err)
 	body, _ := io.ReadAll(resp.Body)
 
-	var servicesResponse api_v3.GetServicesResponse
```
Comment (yurishkuro) [inline cmd/all-in-one/all_in_one_test.go:183]: This was unnecessary coupling. `api_v3` is not internal and not accessible from here.

File: cmd/query/app/apiv3/otlp_translator.go [hunk 3]
```diff
@@ -15,33 +15,13 @@
...
-	b, err := req.MarshalProto()
-	if err != nil {
-		return nil, fmt.Errorf("cannot marshal OTLP: %w", err)
-	}
-	// use api_v3.SpansResponseChunk which has the same shape as otlp.ExportTraceServiceRequest
-	var chunk api_v3.SpansResponseChunk
-	if err := proto.Unmarshal(b, &chunk); err != nil {
-		return nil, fmt.Errorf("cannot marshal OTLP: %w", err)
-	}
-	return chunk.ResourceSpans, nil
```
Comment (yurishkuro) [inline cmd/query/app/apiv3/otlp_translator.go:46]: no triple marshaling!

**Review by jkowall — APPROVED** [review]
LGTM! pretty big PR and chunk of work. Nice job.

**Pull request merged**